src/interface/state_descriptor.hpp

```diff
@@ -249,6 +261,8 @@ class StateDescriptor {
   Dictionary<std::unordered_map<int, Metadata>> sparseMetadataMap_;
   Dictionary<Metadata> swarmMetadataMap_;
   Dictionary<Dictionary<Metadata>> swarmValueMetadataMap_;
+
+  std::vector<std::string> includeWithOutputParams_;
```

Comment:
Maybe use a `std::set` to avoid duplicates? Or get rid of this altogether and just write out all params, as suggested above.

Reply:
Removing this vector with the new commit.